lib/datadog/core/configuration/base.rb

```diff
@@ -34,15 +34,16 @@ def settings(name, &block)
                 value.reset! if value.respond_to?(:reset!)
                 value
               end
-
-              o.type settings_class
             end
+
+            settings_class
```

Comment:
I return the `settings_class` to be able to access the `spec/datadog/core/configuration/base_spec.rb`. Previously the returned value was an `OptionDefintion`, which does not make much sense.